### PR TITLE
[TKW] IGEMM conv merge

### PR DESCRIPTION
**Pull request by Hardcode84**
Merges best IGEMM variant from opt branch.

### CODE REVIEW

File: tests/kernel/wave/wave_e2e_test.py [hunk 1]
```diff
@@ -975,18 +975,21 @@ def test_igemm_conv(n, h, w, c, hf, wf, nf, stride, mem_space, layout, request):
...
+    ratio_m = 2
+    ratio_n = 2
+
     # Expose user-constraints
     constraints: list[tkw.Constraint] = []
-    constraints += [tkw.WorkgroupConstraint(M, BLOCK_M, 0)]
-    constraints += [tkw.WorkgroupConstraint(NF, BLOCK_N, 1)]
-    constraints += [tkw.WaveConstraint(M, BLOCK_M)]
-    constraints += [tkw.WaveConstraint(NF, BLOCK_N)]
+    constraints += [tkw.WorkgroupConstraint(M, BLOCK_M, 1)]
```
Comment (harsh-nod) [inline tests/kernel/wave/wave_e2e_test.py:983]: Does this impact performance? Switching M from 0 to 1 and N from 1 to 0 ?

File: tests/kernel/wave/wave_e2e_test.py [hunk 2]
```diff
@@ -930,15 +930,15 @@ def test_igemm_conv(n, h, w, c, hf, wf, nf, stride, mem_space, layout, request):
         num_iterators=2,
         inputs={
             N: i // SZ_OUT,
-            C: j // (HF * WF),
-            H: (i % SZ_OUT) % W_OUT * stride + (j % (HF * WF)) % WF,
-            W: (i % SZ_OUT) // W_OUT * stride + (j % (HF * WF)) // WF,
+            C: j % C,
```
[the remark this answers is not in the document]
Comment (Hardcode84) [reply tests/kernel/wave/wave_e2e_test.py:934]: added some comment

File: tests/kernel/wave/wave_e2e_test.py (hunk 1 above)
Comment (Hardcode84) [reply tests/kernel/wave/wave_e2e_test.py:984]: It improves perf slightly, but without (contiguous patch it's hard to say)

File: tests/kernel/wave/wave_e2e_test.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (harsh-nod) [reply tests/kernel/wave/wave_e2e_test.py:934]: Thanks.

**Review by harsh-nod — APPROVED** [review]
Lgtm!

**Pull request merged**